test/test_functions

```diff
@@ -1181,3 +1188,125 @@ cp_usrbin() {
+  return 0
+}
+
+
+# creates the default cvmfs.conf file for normal testing
+#
+# @param destination_file  absolute path of the directory
+create_default_cvmfs_config() {
+  local destination_directory="$1"
+  echo 'CVMFS_CACHE_BASE=/root/cvmfs_testing/cache
+CVMFS_RELOAD_SOCKETS=/root/cvmfs_testing/cache
+CVMFS_SERVER_URL=http://cvmfs-stratum-one.cern.ch/cvmfs/@fqrn@
+CVMFS_SHARED_CACHE=no
+CVMFS_HTTP_PROXY=http://ca-proxy.cern.ch:3128
+CVMFS_AUTO_UPDATE=false' > $destination_directory/cvmfs.conf
+}
+
+
+# sets the valgrind options depending on the test type
+#
+# @param test_type  concrete test to execute
+set_valgrind_options() {
+  local CVMFS_OPT_CALLGRIND="--tool=callgrind --dump-before=fuse_session_loop_mt --dump-after=fuse_session_loop_mt --dsymutil=yes --log-file=${FQRN}.callgrind.log -v"
+  local CVMFS_OPT_MEMCHECK="--tool=memcheck --xml=yes --xml-file=${FQRN}.memcheck.xml --trace-children=yes --child-silent-after-fork=yes --leak-check=yes --log-file=${FQRN}.memcheck.log -v"
+  OPTIONS=""
+  case $CVMFS_OPT_TEST_TYPE in
+    callgrind)
+      OPTIONS=$CVMFS_OPT_CALLGRIND
+    ;;
+    memcheck)
+      OPTIONS=$CVMFS_OPT_MEMCHECK
+    ;;
+  esac
+}
+
+
+# creates the environment to run the benchmark
+#
+# @param workspace  absolute path to the directory where the benchmark will put the files in
+setup_benchmark_environment() {
+  local workspace="$1"
+  mkdir -p /cvmfs/$FQRN
+  mkdir -p $CVMFS_OPT_OUTPUT_DIR/$FQRN
+  mkdir -p $workspace/cache
+
+  # create a cvmfs.conf file or use a custom one
+  if [ -z $CVMFS_OPT_CONFIG_FILE ]; then
+    create_default_cvmfs_config $workspace
+  else
+    cp -f $CVMFS_OPT_CONFIG_FILE $workspace
+  fi
+
+  return 0
+}
+
+
+# runs a benchmark $CVMFS_OPT_ITERATIONS times
+run_benchmark() {
+  local count=$CVMFS_OPT_ITERATIONS
+  set_valgrind_options
+
+  # use or not warm cache
+  if [ "$CVMFS_OPT_WARM_CACHE" = "yes" ]; then
+    echo "Starting initial execution"
+    create_warm_cache $workspace
+  else
+    echo "Using cold cache"
+  fi
+
+  echo "Starting loading the profiler and mounting in /cvmfs/${FQRN}"
+  echo "Running valgrind with the following parameters: $OPTIONS"
+  echo "Launching the job $count times"
+  valgrind $OPTIONS cvmfs2 -f -o config=cvmfs.conf,disable_watchdog,simple_options_parsing $FQRN /cvmfs/$FQRN &
+  PID=$!
+  wait_fqrn_mount
+
+  local iteration=1
+  while (( count > 0 )); do
+    echo "Starting iteration ${iteration}"
+    cvmfs_run_benchmark
+    count=$(( count - 1 ))
+    iteration=$(( iteration + 1 ))
+  done
+
+  echo "Iterations finished. Unmounting cvmfs"
+
+  umount /cvmfs/$FQRN
+  wait $PID
+}
+
+
+# collects the files and pastes them into $CVMFS_OPT_OUTPUT_DIR/$FQRN
+collect_benchmark_results() {
+  local benchmark_output_dir="$CVMFS_OPT_OUTPUT_DIR/$FQRN"
+  local workspace=$workdir
+  rm -rf $benchmark_output_dir/*
+  cp  -rf $workdir/$FQRN.*.log $workdir/callgrind.out.* $workdir/$FQRN.memcheck.xml $workdir/cvmfs.conf $benchmark_output_dir > /dev/null 2>&1
```

Comment:
Here, a few more things can be collected.  To my mind comes `uname -a`, `cvmfs_talk -i <fqrn> internal affairs` (the statistics counters), `cvmfs_config stat <fqrn>` (some nicely prepared statistics for the user), `cvmfs2 --version`, the md5 hash of the cvmfs2 binary.